include/netlink/route/link/ip6gre.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * netlink/route/link/ip6gre.h	  IP6GRE interface
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation version 2.1
+ *  of the License.
+ *
+ * Copyright (c) 2018 Thomas Winter <thomas.winter@alliedtelesis.co.nz>
+ */
+
+#ifndef NETLINK_LINK_IP6GRE_H_
+#define NETLINK_LINK_IP6GRE_H_
+
+#include <netlink/netlink.h>
+#include <netlink/route/link.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+	extern int rtnl_link_is_ip6gre(struct rtnl_link *link);
+
+	extern struct rtnl_link *rtnl_link_ip6gre_alloc(void);
+	extern int rtnl_link_ip6gre_add(struct nl_sock *sk, const char *name);
+
+	extern int rtnl_link_ip6gre_set_link(struct rtnl_link *link,  uint32_t index);
+	extern uint32_t rtnl_link_ip6gre_get_link(struct rtnl_link *link);
+
+	extern int rtnl_link_ip6gre_set_iflags(struct rtnl_link *link, uint16_t iflags);
+	extern uint16_t rtnl_link_ip6gre_get_iflags(struct rtnl_link *link);
+
+	extern int rtnl_link_ip6gre_set_oflags(struct rtnl_link *link, uint16_t oflags);
+	extern uint16_t rtnl_link_ip6gre_get_oflags(struct rtnl_link *link);
+
+	extern int rtnl_link_ip6gre_set_ikey(struct rtnl_link *link, uint32_t ikey);
+	extern uint32_t rtnl_link_ip6gre_get_ikey(struct rtnl_link *link);
+
+	extern int rtnl_link_ip6gre_set_okey(struct rtnl_link *link, uint32_t okey);
+	extern uint32_t rtnl_link_ip6gre_get_okey(struct rtnl_link *link);
+
+	extern int rtnl_link_ip6gre_set_local(struct rtnl_link *link, struct in6_addr *local);
+	extern int rtnl_link_ip6gre_get_local(struct rtnl_link *link, struct in6_addr *local);
+
+	extern int rtnl_link_ip6gre_set_remote(struct rtnl_link *link, struct in6_addr *remote);
+	extern int rtnl_link_ip6gre_get_remote(struct rtnl_link *link, struct in6_addr *remote);
+
+	extern int rtnl_link_ip6gre_set_ttl(struct rtnl_link *link, uint8_t ttl);
+	extern uint8_t rtnl_link_ip6gre_get_ttl(struct rtnl_link *link);
```

Comment:
getter should be able to signal an error, like `-NLE_NOATTR` or `-NLE_INVAL`. For that, their signature should always return `int` type, while returning the value as pointer. Yes, that is unfortunately not done consistently, but for new API I think that is the way how it should be done. This means for example:
```C
int rtnl_link_ip6gre_get_ttl(struct rtnl_link *link, uint8_t *out_val)
{
        struct ip6gre_info *ip6gre = link->l_info;

        IS_IP6GRE_LINK_ASSERT(link);

        *out_val = ip6gre->ttl;
        return 0;
}
```

Possibly there is also:
```C
        if (!(ip6gre->ip6gre_mask & IP6GRE_ATTR_TTL))
                return -NLE_NOATTR;
```